spec/factories/service_providers.rb

```diff
@@ -18,5 +18,13 @@
     trait :team do
       association :team, factory: :team
     end
+
+    trait :with_users_team do
+      after(:build) do |service_provider|
+        team = service_provider.user&.teams[0]
```

Comment:
Sorry the goal of this still isn't clear to me, is it just making sure that the SP has a team if it has a user? What is the goal? And it's optional, right?